E2E/e2eTests/Source/Abilities/Sdk.swift

```diff
@@ -0,0 +1,221 @@
+                castor: castor,
+                pluto: pluto,
+                pollux: pollux,
+                mercury: mercury,
+                mediationHandler: BasicMediatorHandler(
+                    mediatorDID: mediatorDID,
+                    mercury: mercury,
+                    store: BasicMediatorHandler.PlutoMediatorStoreImpl(pluto: pluto)
+                )
+            )
+        }
+        
+        func initialize() async throws {
+            try await prismAgent.start()
+            
+            prismAgent.handleReceivedMessagesEvents()
+                .sink(
+                    receiveCompletion: { completion in
+                        switch completion {
+                        case .finished:
+                            print("Publisher completed successfully.")
+                        case .failure(let error):
+                            print("Publisher failed with error: \(error)")
+                        }
+                    },
+                    receiveValue: { message in
+                        if (self.receivedMessages.contains(message.id)) {
+                            return
+                        }
+                        self.receivedMessages.append(message.id)
+                        switch(message.piuri) {
+                        case ProtocolTypes.didcommOfferCredential3_0.rawValue:
+                            self.credentialOfferStack.append(message)
+                        case ProtocolTypes.didcommIssueCredential3_0.rawValue:
+                            self.issueCredentialStack.append(message)
+                        case ProtocolTypes.didcommRequestPresentation.rawValue:
+                            self.proofOfRequestStack.append(message)
+                        default:
+                            break
+                        }
+                    }
+                )
+                .store(in: &cancellables)
+            
+            prismAgent.startFetchingMessages()
+        }
+        
+        func tearDown() async throws {
+            prismAgent.stopFetchingMessages()
+            try await prismAgent.stop()
+        }
+        
+        static private func getPrismMediatorDid() async throws -> DID {
+            let url = URL(string: Config.mediatorOobUrl)!
+            let jsonData: [String: Any] = try await Api.get(from: url)
+            let did = (jsonData["from"] as? String)!
+            return try DID(string: did)
+        }
+        
+        static private func getRootsMediatorDid() async throws -> DID {
+            let url = URL(string: Config.mediatorOobUrl)!
+            let invitationUrl: String = try await Api.get(from: url)
+            let base64data: String = String(invitationUrl.split(separator: "?_oob=").last!)
+            let decodedData = Data(base64Encoded: base64data)!
+            let json = try (JSONSerialization.jsonObject(with: decodedData, options: []) as? [String: Any])!
+            let from = (json["from"] as? String)!
```

Comment:
Just FIY its preferable to use `Codable` and you could use our `Message` structure to parse this. But since this is a test framework its good enough :)

Reply:
Not sure what you mean, can you provide an example?